### PR TITLE
Create datastore for annotations

### LINKED ISSUE #7: Create datastore for annotations
Annotations are metadata for video that occurs at a particular instant in/range of time. Used for recording observations made by people

### PR DESCRIPTION
**Pull request by scott97**
closes #7 

depends on https://github.com/ausocean/openfish/pull/10

### CODE REVIEW

**Review by trekhopton — APPROVED** [review]

File: api/handlers/annotations.go [hunk 1]
```diff
@@ -35,54 +35,110 @@ LICENSE
...
+	}
 
-	// id, _ := ctx.ParamsInt("id", 1)
-	return ctx.JSON("TODO")
+	// Fetch data from the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("Annotation", id)
+	var annotation model.Annotation
+	if store.Get(context.Background(), key, &annotation) != nil {
+		return api.DatastoreReadFailure(err)
```
Comment (saxon-milton) [inline api/handlers/annotations.go:132]: Shouldn't "err" be from the call to store.Get in the if condition ?

File: api/handlers/annotations.go [hunk 2]
```diff
@@ -91,19 +147,110 @@ func GetAnnotations(ctx *fiber.Ctx) error {
...
-		Results: []AnnotationResult{},
+	format := new(api.Format)
+	if err := ctx.QueryParser(format); err != nil {
+		return api.InvalidRequestURL(err)
+	}
+
+	// NOTE: fiber's QueryParser does not handle map[string]string so we need to parse the query manually.
+	// This can be revisited if PR https://github.com/gofiber/fiber/issues/2524 is merged.
+	qry.Observation = make(map[string]string)
+	for k, v := range ctx.Queries() {
```
Comment (saxon-milton) [inline api/handlers/annotations.go:158]: so you can't use the std json package Marshal and Unmarshal functionality ?

File: api/handlers/annotations.go [hunk 3]
```diff
@@ -35,54 +35,110 @@ LICENSE
...
+	id, err := strconv.ParseInt(ctx.Params("id"), 10, 64)
+	if err != nil {
+		return api.InvalidRequestURL(err)
+	}
 
-	// id, _ := ctx.ParamsInt("id", 1)
-	return ctx.JSON("TODO")
+	// Fetch data from the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("Annotation", id)
```
Comment (saxon-milton) [inline api/handlers/annotations.go:129]: We use these keys enough that we should probably create consts for them.

**Review by saxon-milton — CHANGES_REQUESTED** [review]
Just a few things to address.

**Review by saxon-milton — CHANGES_REQUESTED** [review]
Just a few things to address.

File: api/handlers/annotations.go (hunk 2 above)
Comment (scott97) [reply api/handlers/annotations.go:158]: It is not JSON. This is parsing the query parameters in the url

File: api/handlers/annotations.go (hunk 1 above)
Comment (scott97) [reply api/handlers/annotations.go:133]: Fixed

File: api/handlers/annotations.go (hunk 3 above)
Comment (scott97) [reply api/handlers/annotations.go:129]: As in the string ? The key itself is not constant

Comment (scott97) [reply api/handlers/annotations.go:129]: Will make a separate PR updating all three entity types to use consts

**Pull request merged**